Runtime 2.2.0 is not working

## Bug Report

The following code taken from the example docs works on version `2.1.4` but not on version `2.2.0`.

```ts
import { EdgeRuntime } from "edge-runtime";

const initialCode = `
addEventListener('fetch', event => {
  const { searchParams } = new URL(event.request.url)
  const url = searchParams.get('url')
  return event.respondWith(fetch(url))
})`;

const edgeRuntime = new EdgeRuntime({ initialCode });

const response = await edgeRuntime.dispatchFetch("https://example.com");

// If your code logic performs asynchronous tasks, you should await them.
// https://developer.mozilla.org/en-US/docs/Web/API/ExtendableEvent/waitUntil
await response.waitUntil();
```

When running on version `2.2.0` I get the error:

```
TSError: ⨯ Unable to compile TypeScript:
cli2.ts:10:39 - error TS2345: Argument of type '{ initialCode: string; }' is not assignable to parameter of type 'EdgeVMOptions<EdgeContext>'.
  Object literal may only specify known properties, and 'initialCode' does not exist in type 'EdgeVMOptions<EdgeContext>'.

10 const edgeRuntime = new EdgeRuntime({ initialCode });
                                         ~~~~~~~~~~~
cli2.ts:12:36 - error TS2339: Property 'dispatchFetch' does not exist on type 'EdgeVM<EdgeContext>'.

12 const response = await edgeRuntime.dispatchFetch("https://example.com");
                                      ~~~~~~~~~~~~~

    at createTSError (/Users/oscar/Library/pnpm/global/5/.pnpm/ts-node@10.9.1_@types+node@18.0.6_typescript@4.7.4/node_modules/ts-node/src/index.ts:859:12)
    at reportTSError (/Users/oscar/Library/pnpm/global/5/.pnpm/ts-node@10.9.1_@types+node@18.0.6_typescript@4.7.4/node_modules/ts-node/src/index.ts:863:19)
    at getOutput (/Users/oscar/Library/pnpm/global/5/.pnpm/ts-node@10.9.1_@types+node@18.0.6_typescript@4.7.4/node_modules/ts-node/src/index.ts:1077:36)
    at Object.compile (/Users/oscar/Library/pnpm/global/5/.pnpm/ts-node@10.9.1_@types+node@18.0.6_typescript@4.7.4/node_modules/ts-node/src/index.ts:1433:41)
    at transformSource (/Users/oscar/Library/pnpm/global/5/.pnpm/ts-node@10.9.1_@types+node@18.0.6_typescript@4.7.4/node_modules/ts-node/src/esm.ts:400:37)
    at /Users/oscar/Library/pnpm/global/5/.pnpm/ts-node@10.9.1_@types+node@18.0.6_typescript@4.7.4/node_modules/ts-node/src/esm.ts:278:53
    at async addShortCircuitFlag (/Users/oscar/Library/pnpm/global/5/.pnpm/ts-node@10.9.1_@types+node@18.0.6_typescript@4.7.4/node_modules/ts-node/src/esm.ts:409:15)
    at async nextLoad (node:internal/modules/esm/loader:173:22)
    at async ESMLoader.load (node:internal/modules/esm/loader:616:20)
    at async ESMLoader.moduleProvider (node:internal/modules/esm/loader:472:11) {
  diagnosticCodes: [ 2345, 2339 ]
}
```

If I disable typechecking I get the following error proving that the Typescript error is valid:

```
TypeError: edgeRuntime.dispatchFetch is not a function
    at file:///Users/oscar/Desktop/saas-starter2/cli2.ts:14:36
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:541:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
```

I would expect this code to still work but if the arguments have changed it would be good if the documentation could be updated.

--- a/docs/pages/getting-started.mdx
+++ b/docs/pages/getting-started.mdx
@@ -37,7 +37,7 @@ Once it's installed, you can evaluate any script into the runtime context:
 import { EdgeRuntime } from 'edge-runtime'
 
 const runtime = new EdgeRuntime()
-const result = await runtime.evaluate("fetch('https://example.com')")
+const result = await runtime.evaluate("fetch('https://example.vercel.sh')")
 
 console.log(result)
 ```
@@ -67,32 +67,31 @@ const result = await runtime.evaluate("fetch('example.com')")
 console.log(result)
 ```
 
-You can also load some initial code and interact with it as a regular HTTP server:
+You can load some initial code, for example, to be ready for receiving fetch events:
 
 ```js
 import { EdgeRuntime } from 'edge-runtime'
 
 const initialCode = `
 addEventListener('fetch', event => {
-  const { searchParams } = new URL(event.request.url)
-  const url = searchParams.get('url')
-  return event.respondWith(fetch(url))
+  return event.respondWith(fetch(event.request.url))
 })`
 
 const edgeRuntime = new EdgeRuntime({ initialCode })
-
-const response = await edgeRuntime.dispatchFetch('https://example.com')
+const response = await edgeRuntime.dispatchFetch('https://example.vercel.sh')
 
 // If your code logic performs asynchronous tasks, you should await them.
 // https://developer.mozilla.org/en-US/docs/Web/API/ExtendableEvent/waitUntil
 await response.waitUntil()
+
+console.log(await response.text())
 ```
 
-To expose the server:
+and expose it locally to be used via HTTP:
 
 ```js
 import { EdgeRuntime, runServer } from 'edge-runtime'
-import exitHook from 'exit-hook'
+import { onExit } from 'signal-exit'
 import fetch from 'node-fetch'
 
 const initialCode = `
@@ -104,9 +103,13 @@ addEventListener('fetch', event => {
 
 const edgeRuntime = new EdgeRuntime({ initialCode })
 
-const server = await runServer({ runtime: edgeRuntime })
+const server = await runServer({ runtime: edgeRuntime, port: 3000 })
+console.log(`> Edge server running at ${server.url}`)
+onExit(() => server.close())
+```
 
-exitHook(server.close)
+After that, you will be ready to hit your local server from your terminal:
 
-const response = await fetch(server.url)
+```
+curl http://[::]:3000?url=https://example.vercel.sh
 ```

--- a/docs/pages/packages/runtime.mdx
+++ b/docs/pages/packages/runtime.mdx
@@ -49,7 +49,7 @@ addEventListener('fetch', event => {
 const runtime = new EdgeRuntime({ initialCode })
 
 const response = await runtime.dispatchFetch(
-  'http://vercel.com?url=https://example.com'
+  'http://vercel.com?url=https://example.vercel.sh'
 )
 
 // If your code logic performs asynchronous tasks, you should await them.

--- a/packages/runtime/examples/cache.js
+++ b/packages/runtime/examples/cache.js
@@ -3,7 +3,7 @@
 async function handleRequest(event) {
   const cache = await caches.open('default')
   const { searchParams } = new URL(event.request.url)
-  const url = searchParams.get('url') || 'https://example.com'
+  const url = searchParams.get('url') || 'https://example.vercel.sh'
 
   const cacheKey = new URL(url).toString()
   const request = new Request(cacheKey)

--- a/packages/runtime/examples/fetch.js
+++ b/packages/runtime/examples/fetch.js
@@ -65,6 +65,6 @@ async function fetchRequest(url) {
 
 addEventListener('fetch', (event) => {
   const { searchParams } = new URL(event.request.url)
-  const url = searchParams.get('url') || 'https://example.com'
+  const url = searchParams.get('url') || 'https://example.vercel.s'
   return event.respondWith(fetchRequest(url))
 })

--- a/packages/runtime/package.json
+++ b/packages/runtime/package.json
@@ -27,11 +27,11 @@
     "@edge-runtime/format": "workspace:*",
     "@edge-runtime/vm": "workspace:*",
     "async-listen": "2.0.3",
-    "exit-hook": "2.2.1",
     "mri": "1.2.0",
     "picocolors": "1.0.0",
     "pretty-bytes": "5.6.0",
     "pretty-ms": "7.0.1",
+    "signal-exit": "4.0.1",
     "time-span": "4.0.0"
   },
   "devDependencies": {

--- a/packages/runtime/src/cli/index.ts
+++ b/packages/runtime/src/cli/index.ts
@@ -5,7 +5,7 @@ import { promisify } from 'util'
 import { readFileSync } from 'fs'
 import { runServer, type EdgeRuntimeServer } from '../server'
 import childProcess from 'child_process'
-import exitHook from 'exit-hook'
+import { onExit } from 'signal-exit'
 import mri from 'mri'
 import path from 'path'
 
@@ -91,7 +91,7 @@ async function main() {
     }
   }
 
-  exitHook(() => server?.close())
+  onExit(() => server?.close())
   logger(`Waiting incoming requests at ${logger.quotes(server.url)}`)
 }
 

